efixler/scrape
# Auth middleware

This PR enables the first iteration of JWT-based authorization to `scrape` endpoints, via a new middleware.

I'll skip most the usual PR info here in favor of [linking to the documentation updates](https://github.com/efixler/scrape/blob/c7f869fc98fe2afc54e43cbc8d9a9bfafd378e70/README.md#authorization) that step through the process of using authorization and details about how authorization is granted.

## Steps to Test

To test this PR, first follow the steps in the docs, and then issue a request like this:

```
POST http://localhost:8080/extract
Content-Type: application/json
Authorization: Bearer PUT_YOUR_TOKEN_HERE

{
    "url": "https://road.cc/content/tech-news/zipps-new-tyres-ensure-safe-retention-hookless-rims-308387",
    "pp": true
}
```
Also, try it without a token or with a bad token once you've enabled auth.
